scene.go

```diff
@@ -25,6 +25,13 @@ type Scene interface {
 	// Hide is called when an other Scene becomes active
 	Hide()
 
+	// Exit is called when the user or the system requests to close the game
+	// This should be used to cleanup or prompt user if they're sure they want to close
+	// To prevent the default action (close/exit) make sure you set DefaultCloseAction in
+	// your game options to false and MAKE SURE to handle the exiting of the program by calling
+	//     engo.Exit()
+	Exit()
```

Comment:
I think we could, because it's "worse" than just hiding them. So even though it's hidden, they might want to do some cleanup / last-minute saving.